invisible_cities/reco/xy_algorithms.py

```diff
@@ -170,7 +170,11 @@ def corona(pos, qs,
             masked_sipm = []
         # find any masked channel in the first ring around the new local maximum of charge
         neighbours = get_neighbours(new_local_maximum, pitch)
-        n_masked_neighbours = sum(len(list(filter(lambda x: np.allclose(p, x), masked_sipm))) for p in neighbours)
+
+        def number_of_masked_neighbours(neighbours, masked_sipm):
+            return sum(1 for neigh in neighbours for m_sipm in masked_sipm if np.allclose(neigh, m_sipm))
+
+        n_masked_neighbours = number_of_masked_neighbours(neighbours, masked_sipm)
```

Comment:
This still leaves a cryptic nested loop whose meaning is difficult to discern. I was thinking along the lines of

    def my_meaningful_counting_operation_name(sipm):
        return sum(1 for masked_sipm in masked_sipms if np.allclose(neigh, masked_sipm))

    n_masked_neighbours = sum(my_meaningful_counting_operation_name(neigh) for neigh in neighbours)

but after closer inspection, I think that what you actually mean is more like

    def is_masked(sipm):
        return any(np.allclose(sipm, masked_sipm) for masked_sipm in masked_sipms)

    n_masked_neighbours = len(tuple(filter(is_masked, neighbours)))

where an alternative spelling of the last line could be

    n_masked_neighbours = sum(1 for neigh in neighbours if is_masked(neigh))


IIUC your use of `np.allclose` here has the meaning

    def have_same_position_in_3D_space(a,b):
        return np.allclose(a,b)

which would lead to `is_masked` being


    def is_masked(sipm):
        return any(have_same_position_in_3D_space(sipm, masked_sipm) for masked_sipm in masked_sipms)

Reply:
I like this organization and I have implemented it in the code. By the way, I had to look up for 
> IIUC 

in google and the first entry I found was [the following](https://www.iiuc.ac.bd) :-)